django/test/runner.py

```diff
@@ -395,6 +396,12 @@ def __iter__(self):
         return iter(self.subsuites)
 
 
+def convert_test_name_pattern(pattern):
+    if '*' not in pattern:
```

Comment:
This seems to merit an explanation of why Django's behavior should be different than Python's.

Reply:
I'm not sure what should be explained here? Django behavior duplicates Python's because Python does not allow importing `_convert_select_pattern`.

`unittest` uses a function called [`_convert_select_pattern`](https://github.com/python/cpython/blob/02ec92fa7b1dddc23d479ee0b87dc283793505a8/Lib/unittest/main.py#L49-L52) on each supplied pattern. It's automatically called by `argparse` when [performing type conversion from the command line](https://github.com/python/cpython/blob/02ec92fa7b1dddc23d479ee0b87dc283793505a8/Lib/unittest/main.py#L195).

The reason I moved the conversion from `argparse` to `DiscoverRunner.__init__` was to avoid duplication in `runtests.py` and `django/test/runner.py`.

Reply:
The gist of your comment is helpful.